view/frontend/web/internals/common.js

```diff
@@ -450,7 +450,7 @@ requirejs(['algoliaBundle'], function(algoliaBundle) {
 				if (algoliaConfig.instant.enabled && query === '')
 					query = '__empty__';
 
-				window.location = $(this).attr('action') + '?q=' + query;
+				window.location = $(this).attr('action') + '?q=' + encodeURIComponent(query);
```

Comment:
This is already encoded one line 448, @serhiyzhovnir could you explain why you had to redo it here?